builder/azure/chroot/diskattacher.go

```diff
@@ -56,7 +64,13 @@ func (da *diskAttacher) DetachDisk(ctx context.Context, diskID string) error {
 	log.Println("Updating new list of disks attached to VM")
 	err = da.setDisks(ctx, newDisks)
 	if err != nil {
-		return err
+		log.Printf("DetachDisk.setDisks: error: %+v\n", err)
+		log.Println("Checking to see if instance is part of a VM scale set before giving up.")
+		da.ui.Say("Initial call for setting VM instance disks returned an error. Checking to see if instance is part of a VM scale set before giving up.")
+		err = da.setScaleSetDisks(ctx, newDisks)
+		if err != nil {
+			return err
+		}
```

Comment:
So I've walked down this path a bit because it seems like we could be continuing when we should error. Thinking in the case where the VM is not in a scale set and da.setDisks returns an error but for some strange reason the call to `da.setScaleSetDisks()` doesnt. The the original error which was printed to the screen would not return and allow the build to continue up until something downstream fails because the attach disk is not present. 

This case seems unlikely to me as I would expect `da.setScaleSetDisks` to always error when the vm is not in a scale set. But I did want to mention it because it was not immediately clear that something downstream checks for the existence of the said disks and fail appropriately.